src/sql/platform/connection/common/connectionProfileGroup.ts

```diff
@@ -157,6 +157,39 @@ export class ConnectionProfileGroup implements IConnectionProfileGroup {
 		});
 	}
 
+	public removeConnections(connections: ConnectionProfile[]): void {
+		connections.forEach((conn) => {
+			this._childConnections = this._childConnections.filter((curConn) => { return curConn.id !== conn.id; });
+		});
+	}
+
+	public getMatchingConnection(connection: ConnectionProfile): ConnectionProfile | undefined {
+		return this._childConnections.find((conn) => connection.matches(conn));
+	}
+
+	public addConnection(connection: ConnectionProfile): void {
+		const matchingConnection = this.getMatchingConnection(connection);
+		connection.parent = this;
+		connection.groupId = this.id;
+		if (matchingConnection) {
+			const matchingConnectionIndex = this._childConnections.indexOf(matchingConnection);
```

Comment:
Instead of going through childConnections twice, I'd suggest just shortcutting it and finding the index directly instead of calling getMatchingConnection.

(it's basically doing the replaceConnection logic at that point so consider combining those two)